language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.codegen.json
+
+import com.daml.ledger.javaapi.data.Value
+import com.daml.lf.value.json.ApiCodecCompressed
+import com.daml.lf.value.json.ApiCodecCompressed.JsonImplicits._
+import spray.json.JsValue
+
+final class JsonCodec(encodeDecimalAsString: Boolean, encodeInt64AsString: Boolean) {
+  val apiCodec = new ApiCodecCompressed(encodeDecimalAsString, encodeInt64AsString)
+
+  /** Converts `javaapi.data.Value` to `JsValue`. */
+  def toJsValue(value: Value): JsValue =
+    apiCodec.apiValueToJsValue(ValueConversion.toLfValue(value))
```

Comment:
I suggest another function that takes `DefinedDataType[_]` as an argument as well.